app/views/granules/descriptor_document.xml.erb

```diff
@@ -15,8 +15,12 @@
 	<os:Contact><%= ENV['contact']%></os:Contact>
 	<os:Url type="application/atom+xml" rel="results"
 	  params:method="GET"
-	  template="<%= ENV['opensearch_url']%>/granules.atom?datasetId={echo:datasetId?}&amp;shortName=<%= @short_name %>&amp;versionId=<%= @version_id %>&amp;dataCenter=<%= @data_center %>&amp;boundingBox={geo:box?}&amp;lat={geo:lat?}&amp;lon={geo:lon?}&amp;radius={geo:radius?}&amp;geometry={geo:geometry?}&amp;placeName={geo:name?}&amp;startTime={time:start?}&amp;endTime={time:end?}&amp;cursor={os:startPage?}&amp;numberOfResults={os:count?}&amp;offset={os:startIndex?}&amp;uid={geo:uid?}&amp;parentIdentifier={eo:parentIdentifier?}&amp;clientId=<%= @client_id_model.clientId %>">
-	  <params:Parameter name="datasetId" uiDisplay="Collection identifier" value="{echo:datasetId}" title="Inventory associated with a dataset expressed as an ID" minimum="0"/>
+		<% if !@client_id_model.clientId.nil? && !@client_id_model.clientId.blank? -%>
+			template="<%= ENV['opensearch_url']%>/granules.atom?datasetId={echo:datasetId?}&amp;shortName=<%= @short_name %>&amp;versionId=<%= @version_id %>&amp;dataCenter=<%= @data_center %>&amp;boundingBox={geo:box?}&amp;lat={geo:lat?}&amp;lon={geo:lon?}&amp;radius={geo:radius?}&amp;geometry={geo:geometry?}&amp;placeName={geo:name?}&amp;startTime={time:start?}&amp;endTime={time:end?}&amp;cursor={os:startPage?}&amp;numberOfResults={os:count?}&amp;offset={os:startIndex?}&amp;uid={geo:uid?}&amp;parentIdentifier={eo:parentIdentifier?}&amp;clientId=<%= @client_id_model.clientId %>">
+		<% else %>
+	  	template="<%= ENV['opensearch_url']%>/granules.atom?datasetId={echo:datasetId?}&amp;shortName=<%= @short_name %>&amp;versionId=<%= @version_id %>&amp;dataCenter=<%= @data_center %>&amp;boundingBox={geo:box?}&amp;lat={geo:lat?}&amp;lon={geo:lon?}&amp;radius={geo:radius?}&amp;geometry={geo:geometry?}&amp;placeName={geo:name?}&amp;startTime={time:start?}&amp;endTime={time:end?}&amp;cursor={os:startPage?}&amp;numberOfResults={os:count?}&amp;offset={os:startIndex?}&amp;uid={geo:uid?}&amp;parentIdentifier={eo:parentIdentifier?}">
```

Comment:
The bulk of the URL is the same in both if and else. Pull out a variable to handle that.